### PR TITLE
Fix submodule paths when AppSubUrl is not root

### LINKED ISSUE #11002: Link to submodule hosted by the same server via ssh missing port/prefix
- Gitea version (or commit ref): 1.10.0
- Git version:
- Operating system: 
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
If a submodule is hosted by the **same server** as the parent repository and is linked via ssh, the link is missing the prefix/port. 
In case of http linking, the created link is valid. 

The screenshots showing the link for http in the left corner. With ssh the port/prefix is missing.

## Screenshots

http
ssh

### PR DESCRIPTION
**Pull request by zeripath**
Fix #11002

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by guillep2k** [comment]
Unit test failing (related)

**Comment by 6543** [comment]
@zeripath 
```
        	Error Trace:	submodule_test.go:38
        	Error:      	Not equal: 
        	            	expected: "https://localhost/user/repo"
        	            	actual  : "https://localhost/user/repo2/user/repo"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-https://localhost/user/repo
        	            	+https://localhost/user/repo2/user/repo
        	Test:       	TestGetRefURL
```

**Review by 6543 — APPROVED** [review]

File: modules/git/submodule.go [hunk 1]
```diff
@@ -69,16 +66,16 @@ func getRefURL(refURL, urlPrefix, parentPath string) string {
 
 			m := match[0]
 			refHostname := m[2]
-			path := m[3]
+			pth := m[3]
```
Comment (guillep2k) [inline modules/git/submodule.go:69]: 🤣

**Review by guillep2k — APPROVED** [review]

**Comment by 6543** [comment]
ping lgtm

**Pull request merged**